dts/bindings/gpio/gpio-generic.yaml

```diff
@@ -0,0 +1,21 @@
+title: GPIO GENERIC
+
+description: >
+    This is a representation of the GPIO GENERIC nodes
+
+properties:
+    compatible:
+      constraint: "gpio-generic"
+      type: string-array
+      category: required
+      description: compatible strings
```

Comment:
Can simplify this by using `base.yaml` (`dts/binding/base/base.yaml`). Do a `git grep base.yaml` to see examples.